feat: include quantity items received in donations export

scooter-dangle opened the pull request:
Resolves #2376

Note: Modeled after ExportDistributionsCSVService. Next export service
like this might want to DRY things up.

<!--Read comments, before committing pull request read checklist again

# Checklist:

- I have performed a self-review of my own code,
- I have commented my code, particularly in hard-to-understand areas,
- I have made corresponding changes to the documentation,
- I have added tests that prove my fix is effective or that my feature works,
- New and existing unit tests pass locally with my changes ("bundle exec rake"),
- Title include "WIP" if work is in progress.

-->


### Description
<!-- Please include a summary of the change and which issue is fixed. 
Please also include relevant motivation and context.
Guide questions:
  - What motivated this change (if not already described in an issue)?
  - What alternative solutions did you consider?
  - What are the tradeoffs for your solution?
   
List any dependencies that are required for this change. (gems, js libraries, etc.)

Include anything else we should know about. -->

### Type of change

* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?

Tested locally by navigating to the Donations tab and then clicking the `Export Donations` button.

Also added basic spec.

edwinthinks commented on app/controllers/donations_controller.rb:
```diff
@@ -34,7 +34,11 @@ def index
     respond_to do |format|
       format.html
-      format.csv { send_data Donation.generate_csv(@donations), filename: "Donations-#{Time.zone.today}.csv" }
+      # TODO: can we do something fancy here to eagerly load the line
+      # items? (or maybe that's already been done for us?)
```
You can move that eager loading into the service object :). You'd just need to use `includes` from what I recall.

edwinthinks commented on app/services/exports/export_donations_csv_service.rb:
```diff
@@ -0,0 +1,112 @@
+      # Use a where lookup so that I can eager load all the resources
+      # needed rather than depending on external code to do it for me.
+      # This makes this code more self contained and efficient!
+      @donations = donations
```
You should be able to eager load using the same approach as:

https://github.com/rubyforgood/human-essentials/blob/416ea1edec75b7bd462c3e74496a265176be1abb/app/services/exports/export_distributions_csv_service.rb#L7

edwinthinks reviewed (commented):
@scooter-dangle well done! I left a few comments about the eager loading. Let me know if you need any assistance on that.

After addressing that, I think we are good to merge this in :)

scooter-dangle commented:
@edwinthinks, when you get a chance, can you double-check the changes in 334fd9b?

I modified the init to be of the same type as the one used by the Distribution export service, taking IDs instead of model objects.

I also modified the ordering to be `asc` rather than `desc`, which I think is what someone would expect for a date-ordered CSV export.

edwinthinks reviewed (approved):
@scooter-dangle looks great! LGTM!
